core-bundle/src/DependencyInjection/Security/ContaoLoginFactory.php

```diff
@@ -69,4 +88,19 @@ protected function createAuthenticationFailureHandler($container, $id, $config):
     {
         return 'contao.security.authentication_failure_handler';
     }
+
+    private function createTwoFactorPreparationListener(ContainerBuilder $container, string $firewallName): void
+    {
+        $firewallConfigId = TwoFactorFactory::PROVIDER_PREPARATION_LISTENER_ID_PREFIX.$firewallName;
+
+        $container
+            ->setDefinition($firewallConfigId, new ChildDefinition(TwoFactorFactory::PROVIDER_PREPARATION_LISTENER_DEFINITION_ID))
+            ->replaceArgument(3, $firewallName)
+            ->replaceArgument(4, true)
+            ->replaceArgument(5, false)
+            ->addTag('kernel.event_listener', ['event' => 'security.authentication.success', 'method' => 'onLogin', 'priority' => PHP_INT_MAX])
```

Comment:
I'm not sure if it is wise to set the priority to `PHP_INT_MAX`. Nobody will be able to have a listener before this one then in case this should ever be required.

Reply:
That's the 2FA default and has been the case before: https://github.com/scheb/two-factor-bundle/blob/master/Resources/config/security.xml#L70

Reply:
I see. 🤷‍♂